org.lflang/src/org/lflang/generator/ReactionInstance.java

```diff
@@ -356,12 +351,27 @@ public Set<ReactionInstance> dependsOnReactions() {
     public Set<Integer> getLevels() {
```

Comment:
Unless `assignLevels` is invoked in the constructor, there is no assurance that this method will always return a sensible result. Where is `assignLevels` called?

Reply:
I think the problem that was addressed by calling assignLevels, "just in case" is that the level of a reaction is not known until you have constructed the whole reaction graph. So you cannot just do it in the constructor. A simple fix would be to add a simple function to reactorInstance which checks if levels has been assigned yet. (hasLevelsBeenAssigned()) this would walk up the containment hierarchy to the main reactor and check if it has created a reactionInstanceGraph yet (and thus assigned levels). If this for some reason is not true, then I propose to throw an error because the developer might be requesting the levels at a point in time where they are not available (e.g. before the reaction graph has been fully constructed).

How does this sound?

Reply:
OK, that sounds like a good stop-gap measure to me. Still not fond of the original design, but the error will help prevent confusion.